storm-client/src/jvm/org/apache/storm/topology/TopologyBuilder.java

```diff
@@ -353,6 +421,31 @@ public void setStateSpout(String id, IRichStateSpout stateSpout, Number parallel
     }
 
     /**
+     * Define a new spout in this topology.
+     *
+     * @param id the id of this component. This id is referenced by other components that want to consume this spout's outputs.
+     * @param supplier lambda expression that implements tuple generating for this spout
+     * @throws IllegalArgumentException if {@code parallelism_hint} is not positive
+     */
+    public SpoutDeclarer setSpout(String id, SerializableSupplier<Object> supplier) throws IllegalArgumentException {
```

Comment:
make it a generic method
`public <T> SpoutDeclarer setSpout(String id, SerializableSupplier<T> supplier) throws IllegalArgumentException {`

Otherwise following wont work 
```java
        SerializableSupplier<String> supplier = () -> UUID.randomUUID().toString();
        builder.setSpout("spout1", supplier);
```